homeassistant/components/media_player/plex.py

```diff
@@ -136,12 +214,30 @@ def update_devices():
                 continue
 
             if device.machineIdentifier not in plex_clients:
-                new_client = PlexClient(device, plex_sessions, update_devices,
+                new_client = PlexClient(optional_config, device, None,
+                                        plex_sessions, update_devices,
                                         update_sessions)
                 plex_clients[device.machineIdentifier] = new_client
                 new_plex_clients.append(new_client)
             else:
-                plex_clients[device.machineIdentifier].set_device(device)
+                plex_clients[device.machineIdentifier].refresh(device, None)
+
+        # add devices with a session and no client (ex. PlexConnect Apple TV's)
+        if optional_config[CONF_INCLUDE_NON_CLIENTS]:
+            for machine_identifier, session in plex_sessions.items():
+                if machine_identifier not in plex_clients:
```

Comment:
You need to put `and machine_identifier is not None` - this is causing duplicate entities for same players.